<launch>
  <!-- rfid_grid_map tracks an object, grid_map_visualization shows the layer.
        This file requires from a Regions yaml file to be loaded as params -->
<!--
  <env name="ROSCONSOLE_CONFIG_FILE"
        value="$(find rfid_grid_map)/config/custom_rosconsole.conf"/>-->

  <arg name="tagID"                default="390100010000000000000001"/>
  <arg name="object"               default="remote"/>
  <arg name="saveRoute"            default=""/>
  <arg name="loadGrids"            default="false"/>
  <arg name="detectRadius"         default="0.75"/> <!-- detectRadius == b -->
  <arg name="cone_range"           default="1.5"/> <!-- cone_range == a -->
  <arg name="saveTime"             default="60.0"/>
  <arg name="map_resolution"       default="0.3"/>
  <arg name="weight_inc"           default="0.05"/>
  <arg name="weight_dec"           default="0.001"/>
  <arg name="robot_frame"          default="base_link"/>
  <arg name="global_frame"         default="/map"/>
  <arg name="cone_angle"           default="40"/>
  <arg name="temporalDecayPeriod"  default="30"/>
  <arg name="temporalDecayFactor"  default="0.0005"/>
  <arg name="tagToDecayRate"       default="100"/>
  <arg name="with_visualizer"      default="true"/>
   <arg name="mapUpdatePeriod"     default="30.0"/>
   <arg name="probUpdatePeriod"    default="20.0"/>

  <group ns="grid_$(arg tagID)">
  <!-- Launch rfid_grid_map: locate objects -->
  <node pkg="rfid_grid_map" type="rfid_grid_map_node2" name="rfid_grid_map_node" output="screen">
      <remap from="/rfid_grid_map/lastTag" to="/lastTag"/>
      <rosparam param="global_frame">"/map"</rosparam>
      <rosparam param="tagID"                  subst_value="True">"$(arg tagID)"</rosparam>
      <rosparam param="detectRadius"           subst_value="True">"$(arg detectRadius)"</rosparam>
      <rosparam param="map_resolution"         subst_value="True">"$(arg map_resolution)"</rosparam>
      <rosparam param="object"                 subst_value="True">"$(arg object)"</rosparam>
      <rosparam param="saveRoute"              subst_value="True">"$(arg saveRoute)"</rosparam>
      <rosparam param="loadGrids"              subst_value="True">"$(arg loadGrids)"</rosparam>
      <rosparam param="saveTime"               subst_value="True">"$(arg saveTime)"</rosparam>
      <rosparam param="weight_inc"             subst_value="True">"$(arg weight_inc)"</rosparam>
      <rosparam param="weight_dec"             subst_value="True">"$(arg weight_dec)"</rosparam>
      <rosparam param="robot_frame"            subst_value="True">"$(arg robot_frame)"</rosparam>
      <rosparam param="global_frame"           subst_value="True">"$(arg global_frame)"</rosparam>
      <rosparam param="cone_range"             subst_value="True">"$(arg cone_range)"</rosparam>
      <rosparam param="cone_angle"             subst_value="True">"$(arg cone_angle)"</rosparam>
      <rosparam param="temporalDecayPeriod"    subst_value="True">"$(arg temporalDecayPeriod)"</rosparam>
      <rosparam param="temporalDecayFactor"    subst_value="True">"$(arg temporalDecayFactor)"</rosparam>
      <rosparam param="tagToDecayRate"         subst_value="True">"$(arg tagToDecayRate)"</rosparam>
  </node>

  <!-- Launch the grid map visualizer -->
  <node pkg="grid_map_visualization" type="grid_map_visualization" name="grid_map_visualization" output="screen" if="$(arg with_visualizer)">
    <rosparam command="load" file="$(find rfid_grid_map)/config/visualization.yaml" />
    <rosparam param="grid_map_topic" subst_value="True">/grid_$(arg tagID)/rfid_grid_map_node/grid_map</rosparam>
  </node>

  </group>


</launch>
